### PR TITLE
Docker info more robust

### PR DESCRIPTION
**Pull request by pvizeli**
## Proposed change

We should update min requirements to 20.10.0 of docker. Anyway this make it more stable between versions.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by frenck — APPROVED** [review]

**Comment by frenck** [comment]
Simple and effective 👍

File: supervisor/docker/__init__.py [hunk 1]
```diff
@@ -56,10 +56,10 @@ class DockerInfo:
...
         """Create a object from docker info."""
         return DockerInfo(
-            AwesomeVersion(data["ServerVersion"]),
-            data["Driver"],
-            data["LoggingDriver"],
-            data["CgroupVersion"],
+            AwesomeVersion(data.get("ServerVersion", "0.0.0")),
+            data.get("Driver", "unknown"),
+            data.get("LoggingDriver", "unknown"),
+            data.get("CgroupVersion", "1"),
```
Comment (ludeeus) [inline supervisor/docker/__init__.py:62]: 1 is what we want for cgroup version right?
Should we make this "0" to indicate a problem?

Comment (frenck) [reply supervisor/docker/__init__.py:62]: The problem is: It doesn't have to be a problem?

Comment (ludeeus) [reply supervisor/docker/__init__.py:62]: If its not a problem, this PR does not make sense

Comment (pvizeli) [reply supervisor/docker/__init__.py:62]: in case of cgroup, no cgroup means v1

Comment (pvizeli) [reply supervisor/docker/__init__.py:62]: the others can change on docker updates, at least we would start and not break

Comment (pvizeli) [reply supervisor/docker/__init__.py:62]: btw, in future we support cgroup2 as well

Comment (frenck) [reply supervisor/docker/__init__.py:62]: There are 2 things:

1. Docker doesn't provide the info
2. The cgroup version wrong

This mitigates 1 (preventing it from crashing), but it also means we don't know anything about 2. We can't tell if it is wrong or right

**Comment by Joeviocoe** [comment]
This highlights the continued need to disable auto updates for Supervisor.  It should notify for an update, and then allow to SKIP just like HA core and all other addons.

https://community.home-assistant.io/t/how-to-stop-supervisor-auto-update/132271
https://community.home-assistant.io/t/allow-some-degree-of-control-over-supervisor/306830
https://community.home-assistant.io/t/supervisor-option-for-disabling-auto-updates/176334
https://community.home-assistant.io/t/feature-request-block-supervisor-auto-updates/112743

**Comment by frenck** [comment]
> This highlights the continued need to disable auto-updates for Supervisor

This highlight people need to run supported installation methods; as set out by the requirements.

**Pull request merged**